ethreceipts/ethreceipts.go

```diff
@@ -754,6 +754,9 @@ func (l *ReceiptsListener) getMaxWaitBlocks(maxWait *int) uint64 {
 }
 
 func (l *ReceiptsListener) isBlockFinal(blockNum *big.Int) bool {
+	if l == nil || l.latestBlockNum() == nil || blockNum == nil {
+		return false
+	}
 	diff := big.NewInt(0).Sub(l.latestBlockNum(), blockNum)
```

Comment:
It might be better to assign this to a local variable once so that we don't call it twice.

Reply:
ok :) done